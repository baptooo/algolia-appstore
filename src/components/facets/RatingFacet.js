import React from 'react';
import PropTypes from "prop-types";
import Rating from 'react-rating';
import Icon from 'material-ui/Icon';

const RatingFacet = ({ name, values, addOrUpdateFacet }) => {
  let initialValue = parseFloat(values[values.length - 1].name);

  return (
    <Rating
      initialRating={initialValue}
      emptySymbol={<Icon color="disabled">star</Icon>}
      fullSymbol={<Icon color="primary">star</Icon>}
      fractions={2}
      onChange={(rate) => addOrUpdateFacet(name, rate)}
    />
  )
};

RatingFacet.propTypes = {
  name: PropTypes.string.isRequired,
  values: PropTypes.array.isRequired,
  addOrUpdateFacet: PropTypes.func.isRequired
};

export default RatingFacet;