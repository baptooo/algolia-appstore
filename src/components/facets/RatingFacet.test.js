import React from 'react';
import renderer from 'react-test-renderer';
import RatingFacet from './RatingFacet';

const noop = () => {};
const facets = new Array(5).fill({}).map((facet, idx) => ({
  name: idx,
}));

describe('(Component) RatingFacet', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<RatingFacet name="rating" values={facets} addOrUpdateFacet={noop}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});