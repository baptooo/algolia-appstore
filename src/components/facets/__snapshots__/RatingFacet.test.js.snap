// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) RatingFacet should render correctly 1`] = `
Array [
  <span
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
        Object {
            "direction": "ltr",
            "display": "inline-block",
          }
    }
>
    <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
            Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
        }
    >
        <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-1 MuiIcon-colorDisabled-5"
        >
            star
        </span>
        <span
            style={
                Object {
                    "display": "inline-block",
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
            }
        >
            <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-1 MuiIcon-colorPrimary-2"
            >
                star
            </span>
        </span>
    </span>
    <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
            Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
        }
    >
        <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-1 MuiIcon-colorDisabled-5"
        >
            star
        </span>
        <span
            style={
                Object {
                    "display": "inline-block",
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
            }
        >
            <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-1 MuiIcon-colorPrimary-2"
            >
                star
            </span>
        </span>
    </span>
    <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
            Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
        }
    >
        <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-1 MuiIcon-colorDisabled-5"
        >
            star
        </span>
        <span
            style={
                Object {
                    "display": "inline-block",
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
            }
        >
            <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-1 MuiIcon-colorPrimary-2"
            >
                star
            </span>
        </span>
    </span>
    <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
            Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
        }
    >
        <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-1 MuiIcon-colorDisabled-5"
        >
            star
        </span>
        <span
            style={
                Object {
                    "display": "inline-block",
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
            }
        >
            <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-1 MuiIcon-colorPrimary-2"
            >
                star
            </span>
        </span>
    </span>
    <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
            Object {
                "cursor": "pointer",
                "display": "inline-block",
                "position": "relative",
              }
        }
    >
        <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-1 MuiIcon-colorDisabled-5"
        >
            star
        </span>
        <span
            style={
                Object {
                    "display": "inline-block",
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "width": "0%",
                  }
            }
        >
            <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-1 MuiIcon-colorPrimary-2"
            >
                star
            </span>
        </span>
    </span>
</span>,
  <span
    className="MuiTypography-root-7 MuiTypography-caption-17"
>
    (
     results)
</span>,
]
`;
